cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,53 @@
+"""Tests for Openstack cloud volumes"""
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils import testgen
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+
+pytest_generate_tests = testgen.generate([OpenStackProvider], scope='module')
+
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+@pytest.fixture(scope='module')
+def volume_size():
+    return 1
+
+
+@pytest.yield_fixture(scope='function')
+def volume(appliance, provider, volume_size):
+    collection = appliance.collections.volumes
+    volume = collection.create(name=fauxfactory.gen_alpha(),
+                               storage_manager='{} Cinder Manager'.format(provider.name),
+                               tenant=provider.get_yaml_data()['tenant'],
+                               size=volume_size,
+                               provider=provider)
+    yield volume
+
+    try:
+        if volume.exists:
+            volume.delete(wait=False)
+    except Exception:
+        logger.warning('Exception during volume deletion - skipping..')
+
+
+def test_create_volume(volume, volume_size, provider):
+    assert volume.exists
+    assert volume.size == '{} GB'.format(volume_size)
+    assert volume.tenant == provider.get_yaml_data()['tenant']
+
+
+def test_edit_volume(volume):
+    volume.edit(fauxfactory.gen_alpha())
```

Comment:
**Optional**

I'd rather see this name stored and then asserted against after the edit+assert exists

Reply:
Hm, I'm not following you.. New name is recorded inside `edit` method. And then goes assert `volume.exists`

Reply:
if the edit method doesn't work, and doesn't raise exception, you'll never catch it because the name didn't change and it can still be navigated to.

Reply:
Ah.. Well, name is being changed after the flash message assertion, and it is like 99% success. However, you are right